src/hera/workflows/_mixins.py

```diff
@@ -284,3 +285,22 @@ def _build_arguments(self) -> Optional[ModelArguments]:
             parameters=parameters or None,
         )
         return model_arguments
+
+
+class CallableTemplateMixin:
+    def __call__(self, *args, **kwargs) -> SubNodeMixin:
```

Comment:
What do you think about exposing the common Step/Task arguments here? That way users get auto-completion and we can further develop features for the callable, such as template parameter inference

Reply:
Based on Sam's comment in slack:
> We should additionally look into stub generation to better support auto complete

I'd prefer to do that in a new PR whether for 5.0.0 or 5.0.1 so that we can merge this PR as it is 🚀 

(I definitely think getting auto-complete working for callable templates would be a great QOL improvement!)